src/software/new_geom/rectangle.cpp

```diff
@@ -78,14 +78,14 @@ bool Rectangle::contains(const Point &p) const
            p.y() <= negXNegYCorner().y() + diagonal().y();
 }
 
-Point Rectangle::furthestCorner(const Point &p)
+Point Rectangle::furthestCorner(const Point &p) const
 {
     std::vector<Point> corners = points_;
 
-    return *std::max_element(
-        corners.begin(), corners.end(), [&](const Point &corner1, const Point &corner2) {
-            return p.distanceFromPoint(corner1) < p.distanceFromPoint(corner2);
-        });
+    return *std::max_element(corners.begin(), corners.end(),
+                             [&](const Point &corner1, const Point &corner2) {
+                                 return (p - corner1).length() < (p - corner2).length();
```

Comment:
just curious why are we replacing `distanceFromPoint` with a subtraction between two points  and a length check? 

arguably Id say `distanceFromPoint` is a bit more readable

Reply:
I think we should only have one or the other. Which one do you prefer?

Reply:
offline discussion: we are going to do `p1.distance(p2)`

Reply:
fixed it